service/simpleSecretService_test.go

```diff
@@ -0,0 +1,207 @@
+package service
+
+import (
+	"awesomeProject3/service/ciphering"
+	"errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+const notSecretVal = "decryptedValue"
+const secretVal = "encryptedValue"
+const valFromStorage = "valueFromStorage"
+const password = "passwordIs32bytelongandOtherTxt"
+const read = "Read"
+
+type mockStorage struct{ mock.Mock }
+
+func newMockStorage() *mockStorage { return &mockStorage{} }
+
+func (m *mockStorage) Read(key string) (string, error) {
+	arg := m.Called(key)
+	return arg.String(0), arg.Error(1)
+}
+
+func (m *mockStorage) Write(key string, value string) error {
+	arg := m.Called(key, value)
+	return arg.Error(0)
+}
+
+type mockEncoder struct{ mock.Mock }
+
+func newMockEncoder() *mockEncoder { return &mockEncoder{} }
+
+func (m *mockEncoder) Encrypt(plaintext string, cipherKey []byte) (string, error) {
+	arg := m.Called(plaintext, cipherKey)
+	return arg.String(0), arg.Error(1)
+}
+func (m *mockEncoder) Decrypt(ct string, cipherKey []byte) (string, error) {
+	arg := m.Called(ct, cipherKey)
+	return arg.String(0), arg.Error(1)
+}
+
+func TestSimpleSecretService_WriteSecret2(t *testing.T) {
+	var testCases = []struct {
+		name           string
+		encryptVal     string
+		encryptErr     error
+		value          string
+		writeErr       error
+		secretKey      string
+		key            string
+		numberOfCallsW int
+		numberOfCallsE int
+		expectedErr    bool
+	}{
+		{
```

Comment:
I believe that several use cases are missing here. Just at first glance:
- if cannot decode key
- if cannot write to storage

Investigate for more cases (if applicable).